src/host/_stream.cpp

```diff
@@ -410,24 +227,24 @@ try
 
                 textBuffer.GetRowByOffset(pos.y).SetWrapForced(false);
                 pos.y = pos.y + 1;
-                AdjustCursorPosition(screenInfo, pos, keepCursorVisible, psScrollY);
+                AdjustCursorPosition(screenInfo, pos, interactive, psScrollY);
                 continue;
             }
             case UNICODE_CARRIAGERETURN:
             {
                 auto pos = cursor.GetPosition();
                 pos.x = 0;
-                AdjustCursorPosition(screenInfo, pos, keepCursorVisible, psScrollY);
+                AdjustCursorPosition(screenInfo, pos, interactive, psScrollY);
                 continue;
             }
             default:
                 break;
             }
 
-            if (WI_IsFlagSet(dwFlags, WC_INTERACTIVE) && IS_CONTROL_CHAR(*it))
+            if (interactive && *it < L' ')

```

Comment:
Maybe add a comment to clarify that `< ' '` means it's a control character (which I didn't immediately think of when reading this)